AspNetCoreApplication/Controller/AuthorController.cs

```diff
@@ -16,58 +17,37 @@ namespace AspNetCoreApplication.Controller
     [Route("api/authors")]
     public class AuthorController : ControllerBase
     {
-        private readonly DataContext dataContext;
-
-        public AuthorController(DataContext dataContext)
+        private readonly IRepository<Author> author;
+        public AuthorController(IRepository<Author> author)
         {
-            this.dataContext = dataContext;
+            this.author = author;
         }
-
         [HttpGet]
-        public async Task<List<AuthorItem>> Get()
+        public Task<List<AuthorItem>> Get()
         {
-            var authors = await dataContext.Authors.ToListAsync();
-            return authors.Select(x => x.ConvertTo<AuthorItem>()).ToList();
+            return author.Get<AuthorItem>();
         }
 
         [HttpGet("{id}")]
-        public async Task<AuthorDetail> Get(int id)
+        public Task<AuthorDetail> Get(int id)
         {
-            var author = await dataContext.Authors.FindAsync(id) ??
-                         throw new NotFoundException("Author can't be found");
-
-            return author.ConvertTo<AuthorDetail>();
+            return author.Get<AuthorDetail>(id);
         }
         [HttpPost]
-        public async Task Add([FromBody] AuthorForm authorForm)
+        public Task Add([FromBody] AuthorForm authorForm)
```

Comment:
```suggestion
        public async Task Add([FromBody] AuthorForm authorForm) => await author.Add(authorForm);
```

Đưa thao tác convert nào bên trong repository

Reply:
done